docs/proposals/operator-sdk-v1-migration.md

```diff
@@ -0,0 +1,285 @@
+`shipwright.io` (or subdomain therein). The latter would be a breaking change, forcing current
+users to migrate their existing Build and BuildStrategy instances.
+
+The project does not require multi-group APIs at present, and moving forward this will likely
+remain the case [3].
+
+[2] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#create-a-new-project
+[3] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#check-if-your-project-is-multi-group
+
+
+### API Migration
+
+Operator SDK v1.x changed the package layout of APIs from 
+`pkg/apis/<group>/<version>/<kind>_types.go` to `/api/version/<kind>_types.go`. Other notable
+changes to API generation include [4]:
+
+1. Replacing `+k8s:deepcopy-gen:interfaces=...` markers with `+kubebuilder:object:root=true`.
+2. Removing `// +k8s:openapi-gen=true` and other related openapi markers. This can be added back,
+   but is not necessary if we do not use `openapi-gen`.
+
+We currently use `deepcopy-gen` markers to generate our deep copy code, and use
+`k8s.io/code-generator` to create our OpenAPI generated code under the covers. Presumably using
+the `kubebuilder` marker will let us retain our generated deep copies.
+
+[4] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#apis
+
+### Controllers
+
+Controller code can more or less be migrated as is from
+`pkg/controller/<kind>/<kind>_controller.go` to `/controllers/<kind>_controller.go` [5]. There are
+a few subtleties with generated types and a change to the manager interface, but in general the
+shift is straightforward.
+
+[5] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#controllers
+
+### RBAC
+
+RBAC is maintained via kubebuilder RBAC Markers, rather than separate YAML files. The RBAC markers
+are placed on the `Reconcile` method of each controller, alongside implementing code. SDK projects
+can then generate RBAC for the controllers via a make target (`make manifests`). RBAC YAML has been
+moved to `config/rbac/` directory, rather than living alongside deployment manifests in `deploy`.
+
+New operators are cluster-scoped. We believe this is the right scoping for the build controllers.
+
+[6] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#set-the-rbac-permissions
+[7] https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/
+
+### Migrate main.go
+
+The main invocation has changed its default generated code with respect to leader election.
+Operator SDK has not deprecated `leader.Become` (yet), but it has moved new code to use the leader
+election libraries from controller-runtime [8].
+
+If we maintain usage of `leader.Become`, there are changes to default behavior which we need to
+account for:
+
+1. The default metrics endpoint now binds to port `:8080` instead of `:8383`. To use the old port,
+   we need to pass the `--metrics-addr` argument to our binary.
```

Comment:
I wouldn't suggest passing --metrics-addr 8383 , I would enforce usage of the new metrics port if we were to migrate.

Reply:
new port looks fine to me